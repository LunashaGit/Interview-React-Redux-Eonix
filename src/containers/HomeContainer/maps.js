import { createStructuredSelector } from "reselect";

import { makeSelectHomeContainerCounter } from "./selector";
import { incrementAction, decrementAction } from "./action";

export const mapStateToProps = createStructuredSelector({
  counter: makeSelectHomeContainerCounter(),
});

export function mapDispatchToProps(dispatch) {
  return {
    dispatch,
    onIncrement: () => dispatch(incrementAction()),
    onDecrement: () => dispatch(decrementAction()),
  };
}
