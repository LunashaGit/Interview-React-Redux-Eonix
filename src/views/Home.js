import React from "react";

function Home({ onDecrement, onIncrement, counter }) {
  return (
    <div className="counter">
      <h1>Counter: {counter}</h1>
      <button onClick={onIncrement}>Increment</button>
      <button onClick={onDecrement}>Decrement</button>
    </div>
  );
}

export default Home;
