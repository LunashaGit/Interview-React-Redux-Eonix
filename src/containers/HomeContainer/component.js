import React from "react";
import Home from "../../views/Home";
import { useInjectReducer } from "../../utils/injectReducer";
import reducer from "./reducer";

const key = "homeContainer";

export default function HomeContainer(props) {
  useInjectReducer({ key, reducer });

  return <Home {...props} />;
}
