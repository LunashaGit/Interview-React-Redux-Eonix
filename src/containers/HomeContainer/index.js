import { connect } from "react-redux";
import { compose } from "redux";
import { mapStateToProps, mapDispatchToProps } from "./maps";

import HomeContainer from "./component";

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withConnect)(HomeContainer);
