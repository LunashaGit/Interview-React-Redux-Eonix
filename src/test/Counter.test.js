import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Home from "../views/Home";
import { Provider } from "react-redux";
import configureStore from "../configureStore";
import history from "../utils/history";
import HomeContainer from "../containers/HomeContainer";

configure({ adapter: new Adapter() });
const initialState = {};
const store = configureStore(initialState, history);

describe("Home component", () => {
  it("should render without throwing an error", () => {
    expect(
      shallow(
        <Provider store={store}>
          <Home />
        </Provider>
      ).exists()
    ).toBe(true);
  });

  // Increment
  it("should increment count", () => {
    const wrapper = shallow(<Home />);
    wrapper.find("button").at(0).simulate("click");
    expect(wrapper.find("h1").text()).toEqual("Counter: ");
  });

  // Decrement
  it("should decrement count", () => {
    const wrapper = shallow(<Home store={store} />);
    wrapper.find("button").at(1).simulate("click");
    expect(wrapper.find("h1").text()).toEqual("Counter: ");
  });
});
