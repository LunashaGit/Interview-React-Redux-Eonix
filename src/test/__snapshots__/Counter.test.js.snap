exports[`Home component renders correctly 1`] = `undefined`;
